### PR TITLE
Only use locks when GPORCA is in parallel mode

### PR DESCRIPTION
**Pull request by vraghavan78**

### CODE REVIEW

File: libnaucrates/src/statistics/CStatistics.cpp [hunk 1]
```diff
@@ -2882,7 +2882,11 @@ CStatistics::UlIndexUpperBoundNDVs
 {
 	GPOS_ASSERT(NULL != pcr);
  	CAutoMutex am(m_mutexCardUpperBoundAccess);
```
Comment (hsyuan) [inline libnaucrates/src/statistics/CStatistics.cpp]: this should go to if block

Comment (vraghavan78) [inline libnaucrates/src/statistics/CStatistics.cpp]: i do not think so.

Comment (hsyuan) [inline libnaucrates/src/statistics/CStatistics.cpp]: I take it back. I was wrong. If we put it inside if block, CAutoMutex will be destroyed after we go outside of if block, then the lock will not take effective.

**Comment by hsyuan** [comment]
Oh, just FYI: If you want to bump the version, bump the patch version, don't bump the minor version. Because there is no API changes.

**Comment by oarap** [comment]
-1 
I do not like this approach. We need to decide how we do this first. This is so `if`fy.

**Comment by hsyuan** [comment]
How about this:

```c++
#define GPOS_LOCK(am)  \
	do { \
		if (GPOS_FTRACE(EopttraceParallel)) \
			am.Lock(); \
	} while(0)
```

So we can just replace `am.Lock();` with `GPOS_LOCK(am)`.
lol

**Pull request closed without merging**